### ISSUE TITLE
Can't install GPU operator on virtualized OCP 4.6 on VMWare ESXi 7.0.1 with direct attached V100 GPU

### ISSUE THREAD

**Issue by chuempel** [NVIDIA/gpu-operator#171]
Hello,

the installation of the GPU operator 1.6.2 via Operator Hub on a virtualized OCP 4.6.19 worker node running on VMWare ESXi 7.0.1 with single V100 32GB GPU attached via PCIe passthrough fails and brings the ESXi host down.

Is there a specific setup of the worker node VM and/or the attachment of the V100 GPU needed in vSphere ESXi in order to get the installation of the GPU operator running ok?

Thanks
C.H.

Environment:
- OpenShift 4.6.19, Kubernetes v1.19.0+8d12420
- VMWare ESXi 7.0.1
- 1x Xeon Gold 6246R, 16c / 32t, 64 GB RAM
- 1x Tesla V100 GPU with 32GB VRAM for PCIe-based

Steps to reproduce the issue
- Following the instructions from here: [https://docs.nvidia.com/datacenter/kubernetes/openshift-on-gpu-install-guide/index.html#openshift-gpu-support-install-via-operatorhub](url)
- Deploy GPU operator 1.6.2 on OCP cluster
- Create Cluster Policy gpu-cluster-policy
- Wait 4 minutes -> Kernel crash of ESXi will appear.

Additional information:

![esxi-1](https://user-images.githubusercontent.com/67739233/111899914-d1a0bb80-8a2f-11eb-8f7a-66f2e0563e5a.jpg)
![esxi-2](https://user-images.githubusercontent.com/67739233/111899916-d7969c80-8a2f-11eb-9d67-b6a69fc927b4.jpg)

**Comment by shivamerla** [comment]
@chuempel Not sure if the PSOD is actually caused by driver installation within the guest. Did you open a support ticket with VMware to look into the PSOD issue?

**Comment by shivamerla** [comment]
Also, can you provide details about server type/model and if its certified for use with V100?

**Comment by chuempel** [comment]
@shivamerla we are going to open a support ticket with VMware.
For the server type/model:
- Fujitsu RX2540 M5
- 1x Intel Xeon Gold 6246R 16c 3.4 GHz
- 2x 32 GB 2RX4 DDR4-2933 R ECC
- 2x HD SAS 12G 600GB 10k 512n Hot Plug
- PRAID CP400i
- Risercard PCIe 2 FH Slots x16/x8 right
- 2x X710-DA2 2x 10Gb SFP+
- 1x PY NVIDIA Tesla V100 32GB for PCIe-based

**Comment by ticlazau** [comment]
@chuempel this is more a VMware ESXi issue. Recent CPUs from Intel and AMD implement a machine-check architecture that detects and reports hardware issues, including system bus errors, RAM (ECC and parity) errors, and other CPU errors. There are a set of model-specific registers (MSRs) that are used to report errors. When a hardware error occurs, global and bank-specific status machine-check architecture registers are populated with information regarding the cause, and whether the CPU can safely continue execution. In the case of a correctable error, ESXi reports the incident and register contents in the VMkernel logs. If an error is uncorrectable, and the CPU cannot continue safely, ESXi halts with a purple diagnostic screen.

Read the crash log from ESXi by following this procedure: https://kb.vmware.com/s/article/1006796 

Rgds,
F.M.

**Comment by chuempel** [comment]
@ticlazau I have forwarded your information on reading out the crash log from ESXi. We were able to test that V100 is working ok, by using a CentOS VM and installing the NVIDIA drivers by hand there. The key to get this working was to have the VM boot in EFI / UEFI mode. In our OCP GPU worker node VM we were usging BIOS which is not supported with DirectPathThroughIO
See this link: https://blogs.vmware.com/apps/2018/09/using-gpus-with-virtual-machines-on-vsphere-part-2-vmdirectpath-i-o.html
We are going now to retest with a Bare Metal RHCOS GPU worker node setup to validate that the GPU operator is working here.
